### PR TITLE
Formula

### PR DESCRIPTION
**Pull request by obennet**
works for value drivers and external factors

### CODE REVIEW

**Comment by obennet** [comment]
Failing tests are the ones connected to running simulation with files. Same as others PR

File: sedbackend/apps/cvs/simulation/storage.py [hunk 1]
```diff
@@ -447,6 +454,7 @@ def get_market_values(db_connection: PooledMySQLConnection, vcs_row_id: int, vcs
 
 
 # TODO fix how the formulas are parsed
```
Comment (EppChops) [inline sedbackend/apps/cvs/simulation/storage.py:456]: Shouldn't leave old code commented out

File: sedbackend/apps/cvs/simulation/storage.py [hunk 2]
```diff
@@ -463,6 +471,27 @@ def parse_formula(formula: str, vd_values, mi_values):
...
+        for name in vd_names:
+            unit = vd["unit"] if vd["unit"] is not None and vd["unit"] != "" else "N/A"
+            if name == f'{vd["name"]} [{unit}]':
+                new_formula = expr.replace_prefix_variables("VD", name, str(vd["value"]), new_formula)
+    for mi in mi_values:
+        for name in mi_names:
+            unit = mi["unit"] if mi["unit"] is not None and mi["unit"] != "" else "N/A"
+            if name == f'{mi["name"]} [{unit}]':
+                new_formula = expr.replace_prefix_variables("EF", name, str(mi["value"]), new_formula)
+    new_formula = expr.remove_strings_replace_zero(new_formula)
```
Comment (EppChops) [inline sedbackend/apps/cvs/simulation/storage.py:492]: I don't know if this line but something in the parsing does not work right.

Got the following error with the given input:
Screenshot_2023-04-05_22-21-57
Screenshot_2023-04-05_22-22-10

**Review by EppChops — CHANGES_REQUESTED** [review]
The parser seems to always change the VD's and the EF's to 0, no matter what. See attached screen shots of formulas and sim results. Please fix parsing <3


It does run tho ;)

Screenshot_2023-04-05_22-26-05
Screenshot_2023-04-05_22-26-14
Screenshot_2023-04-05_22-27-12

**Comment by EppChops** [comment]
Hi!

I hate to tell you but the parsing still doesn't work correctly. I have two examples, the first ones with no division by zero, and the last one with division by zero which gives error "Could not parse formula":

Example 1:
Screenshot_2023-04-12_12-15-54
Screenshot_2023-04-12_12-19-52
Screenshot_2023-04-12_12-20-06
Screenshot_2023-04-12_12-21-25

Example 2:
Screenshot_2023-04-12_12-22-52
Screenshot_2023-04-12_12-23-06
Screenshot_2023-04-12_12-23-15

Final example with divison by zero (rather, no values in the VD's);
Screenshot_2023-04-12_12-28-59

**Comment by EppChops** [comment]
Found another thing. 

With the following formulas: 
Screenshot_2023-04-12_13-02-02


The simulation does work when no values on VD's are set, as can be seen here: 
Screenshot_2023-04-12_13-00-28
Screenshot_2023-04-12_13-00-51


But it does not work when I put values to my VD's:
Screenshot_2023-04-12_13-01-18
Screenshot_2023-04-12_13-01-32

**Comment by EppChops** [comment]
LGTM good job!

**Pull request merged**